src/main/java/com/novemberain/quartz/mongodb/JobCompleteHandler.java

```diff
@@ -32,22 +32,29 @@ public JobCompleteHandler(TriggerAndJobPersister persister, SchedulerSignaler si
     }
 
     public void jobComplete(OperableTrigger trigger, JobDetail job,
-                            CompletedExecutionInstruction executionInstruction)
-            throws JobPersistenceException {
+                            CompletedExecutionInstruction executionInstruction) {
         log.debug("Trigger completed {}", trigger.getKey());
 
         if (job.isPersistJobDataAfterExecution()) {
             if (job.getJobDataMap().isDirty()) {
                 log.debug("Job data map dirty, will store {}", job.getKey());
-                jobDao.storeJobInMongo(job, true);
+                try {
+                    jobDao.storeJobInMongo(job, true);
+                } catch (JobPersistenceException e) {
+                    throw new RuntimeException(e);
```

Comment:
@pwojnowski are you OK with this change? @dorogush may I ask what's the idea here?